feat(api): access response properties from api errors (#6775)
aws-amplify/docs: src/fragments/lib/restapi/js/fetch.mdx

</Callout>

## Access HTTP response from errors

The REST API handler may throw an `ApiError` error instance. In case the error is caused by an HTTP response with
non-2xx status code, the error instance will provide a `response` property. The `response` property contains following
properties:
* `statusCode`: HTTP status code
* `headers`: HTTP response headers
* `body`: HTTP response body as a string

The following example shows how to access the HTTP response from an `ApiError` instance, so that you can handle the 
error response from your REST API endpoint:

```javascript
import { ApiError } from 'aws-amplify/api';

try {

>   await restOperation.response;

Reviewer: maybe be a bit more explicit here with restOperation by adding a line like:

```js
const restOperation = get('/...')
```

Reviewer: will probably also need to update the import to include "get"

</Callout>

## Access HTTP response from errors

> The REST API handler may throw an `ApiError` error instance. In case the error is caused by an HTTP response with

Reviewer 2: ```suggestion
The REST API handler may throw an `ApiError` error instance. If the error is caused by an HTTP response with a
```
